### DISCUSSION TITLE
How to debug sycl program using lldb or gdb ?

### DISCUSSION THREAD

**Discussion by wangzy0327** [intel/llvm#12919] [Q&A]
### Describe the bug

How to debug sycl program using lldb or gdb ?

### Environment

Os:Linux
Version:clang-16

**Comment by KornevNikita** [comment]
Converted to discussion according to the description.

**Comment by wangzy0327** [comment]
@KornevNikita Can you help me about the question?

**Reply by KornevNikita** [reply]
@wangzy0327 sorry I don't have much experience with debugging kernel code. I know there is [oneapi-gdb](https://www.intel.com/content/www/us/en/developer/tools/oneapi/distribution-for-gdb.html#gs.62fhuc), that might be [helpful](https://www.intel.com/content/www/us/en/docs/distribution-for-gdb/tutorial-debugging-dpcpp-linux/2023-0/debug-a-sycl-application-on-a-gpu.html).

@AlexeySachkov

**Reply by wangzy0327** [reply]
When I use gdb to debug program which by SYCL clang -g compile ，it display ```Dwarf Error: DW_FORM_strx1 found in non-DWO CU [in module /home/wzy/sycl_workspace/main-clang.out]
(No debugging symbols found in main-clang.out)```, even it is host-side code.
Can you give me some help ?
